### DISCUSSION TITLE
Polaris社区新手入门课程策划

### DISCUSSION THREAD

**Discussion by andrewshan** [polarismesh/polaris#316] [Ideas]
# 系列课程

## 形式

以持续的一系列课程形式展现，覆盖微服务开发者从入门到精通的成长之路。
课程介质包括视频课程，以及文字课程两类。
课程时间大概30分钟左右，最多不超过40分钟。

课程分为3大类：

快速入门：主要是给予用户指南，可以快速接入使用北极星
使用指南：介绍北极星的各项功能，用户可以按照指南使用
架构原理：介绍北极星的各部分系统架构，以及源码走读

## 发布平台

B站、知乎、视频号、CSDN、腾源会、腾讯课堂、极客时间

## 完成时间

快速入门课程：预计4月底完成

使用指南课程：预计6月底完成

架构原理课程：预计8月底完成

### 第一部分：快速入门

第1章 背景

- [北极星简介](https://www.bilibili.com/video/BV1A34y147tx?spm_id_from=333.999.0.0)
- [安装服务端](https://www.bilibili.com/video/BV11Z4y1m7Lh/?spm_id_from=333.788.recommend_more_video.1)

第2章 Java 微服务开发

- [polaris-java 介绍](https://www.bilibili.com/video/BV1x3411T7jU/?spm_id_from=333.788.recommend_more_video.2)
- 使用 grpc-java with polaris
- [使用 spring-cloud with polaris](https://www.bilibili.com/video/BV163411P7BT/?spm_id_from=333.788.recommend_more_video.5)
- 使用 spring-boot with polaris
- spring cloud 如何无感知接入
- dubbo 如何无感知接入

第3章 Go 微服务开发

- [polaris-go介绍](https://www.bilibili.com/video/BV1ML4y1V7tM/?spm_id_from=333.788.recommend_more_video.-1)
- [使用 grpc-go with polaris](https://www.bilibili.com/video/BV1VY411j7ty?spm_id_from=333.337.search-card.all.click)
- 使用 go-zero with polaris
- 使用 go-frame with polaris【待定】

第4章 C++ 及其他语言开发

- [polaris-cpp 介绍](https://www.bilibili.com/video/BV1h54y1o7G1?spm_id_from=333.999.0.0)
- polaris-php 介绍
- polaris-lua 介绍

第5章 DNS 和 Proxy 网格

- [使用内网域名解析](https://www.bilibili.com/video/BV1kZ4y1b7fS?spm_id_from=333.999.0.0)
- 使用 envoy 接入

第6章 API网关接入

- 使用 nginx with polaris

### 第二部分：使用指南

第1章 配置管理

- 配置管理简介及使用

第2章 流量调度

- 如何实现全链路灰度
   - 使用spring cloud实现全链路灰度
   - 使用dubbo实现全链路灰度
- 如何实现就近路由

第3章 故障容错

- 故障容错介绍
- 面向服务的实例熔断和切换
- 服务熔断和降级【待定】
- 面向接口的实例熔断和切换【待定】
- 接口熔断和降级【待定】

第4章 限流

- 限流简介

第5章 用户权限控制

- 用户权限控制介绍

### 第三部分：原理解析

- [第6章 服务端代码架构解读](https://www.bilibili.com/video/BV1uY411g7ap?spm_id_from=333.999.0.0)

**Comment by houseme** [comment]
可以在readme中添加各个平台的入口，咱们社区也可以做个每个视频的汇总入口

**Reply by andrewshan** [reply]
可以的，这个建议很好

**Comment by songoow** [comment]
感觉每期视频，可以给用户提供一个可以复现的demo代码:smiley:，或者类似的demo案例演示

**Reply by chuntaojun** [reply]
感觉是可以，可以创建一个 polaris-learning 库，这个库里面记录着每期的相关学习链接、demo 代码

**Reply by andrewshan** [reply]
demo是不是统一归档到polaris-examples里面比较好？

视频的链接，我们直接在官网文档中给出来？

**Reply by chuntaojun** [reply]
> demo是不是统一归档到polaris-examples里面比较好？
> 
> 视频的链接，我们直接在官网文档中给出来？

也是可以的

**Comment by lepdou** [comment]
赞~

**Comment by andrewshan** [comment]
需要讨论的问题：
1.  各视频网站的账号的创建?
p_wangying。
视频号：ready
B站、知乎，今天开始提交创建

2.  视频如何录制，怎么进行剪辑？
开发输出视频，wangying帮忙推到各个视频网站。
使用OBS来进行录制

4.  每月完成多少个课程？
每个月平均每人8个课程（每周1小时投入，每周输出2个，前期熟悉阶段可以每周1个），搭配文档进行补齐。
重点先把入门和进阶的补上

5.  课程的开发分工
服务治理：andrew, springliao
配置，spring cloud, spring boot：ledou, sky

**Comment by chuntaojun** [comment]
课程认领（暂定）

02.10 Polaris入门-dubbo框架接入

02.11 Polaris入门-gozero框架接入

04.2 Polaris进阶-使用规则路由

04.3 Polaris进阶-使用元数据路由

04.4 Polaris进阶-使用就近路由

07.1 Polaris进阶-北极星鉴权体系

08.7 Polaris高阶-GoSDK/JavaSDK/C++SDK服务路由实现

**Comment by zilongTong** [comment]
👍🏻

**Comment by yanshinian** [comment]
也可以把视频放到油管上。

**Comment by fuxshen** [comment]
官方视频方面的建议：
         1、希望在日后如果有版本更新的话，能及时将新版本的特性和具体整合操作及时更新
         2、希望能够有北极星面对微服务架构出现的问题，所采用的解决方案的具体介绍和实际落地的整合操作
         3、希望视频能够“接地气”一些，能通俗易懂的让大家“阅读”后，快速落地实践
         4、对应版本或者某些特性的源码以及流程的讲解
         5、对应的视频的学习资料
